frame/balances/src/lib.rs

```diff
@@ -219,6 +219,7 @@ pub mod pallet {
 			+ FixedPointOperand;
 
 		/// Handler for the unbalanced reduction when removing a dust account.
+		#[pallet::no_default]
```

Comment:
```
diff --git a/frame/examples/basic/src/lib.rs b/frame/examples/basic/src/lib.rs
index 706a088fa97..6946449b35c 100644
--- a/frame/examples/basic/src/lib.rs
+++ b/frame/examples/basic/src/lib.rs
@@ -367,10 +367,9 @@ pub mod pallet {
        pub trait Config: pallet_balances::Config + frame_system::Config {
                // Setting a constant config parameter from the runtime
                #[pallet::constant]
-               #[pallet::no_default]
                // It is very unfortunate that we cannot have this have a default either, because it relies
                // on `<Self as pallet_balances::Config>`
-               type MagicNumber: Get<Self::Balance>;
+               type MagicNumber: Get<<Self as pallet_balances::Config>::Balance>;
 
                /// The overarching event type.
                type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
diff --git a/frame/support/procedural/src/pallet/expand/config.rs b/frame/support/procedural/src/pallet/expand/config.rs
index baa5ae74cdd..d2f1c792f5a 100644
--- a/frame/support/procedural/src/pallet/expand/config.rs
+++ b/frame/support/procedural/src/pallet/expand/config.rs
@@ -50,9 +50,12 @@ pub fn expand_config(def: &mut Def) -> TokenStream {
                        Ok(krate) => krate,
                        Err(err) => return err.to_compile_error(),
                };
+
+               let super_traits = config_item.supertraits.iter();
+
                quote!(
                        #[#support::macro_magic::export_tokens]
-                       pub trait DefaultConfig {
+                       pub trait DefaultConfig: #( #super_traits )+* {
                                #(#trait_items)*
                        }
                )
diff --git a/frame/support/procedural/src/pallet/parse/config.rs b/frame/support/procedural/src/pallet/parse/config.rs
index b42b06d6aaa..a662b756b5b 100644
--- a/frame/support/procedural/src/pallet/parse/config.rs
+++ b/frame/support/procedural/src/pallet/parse/config.rs
@@ -58,6 +58,8 @@ pub struct ConfigDef {
        /// No, if `None`.
        /// Yes, if `Some(_)`, with the inner items that should be included.
        pub default_sub_trait: Option<Vec<syn::TraitItem>>,
+
+       pub super_traits: Vec<syn::TypeParamBound>,
 }
 
 /// Input definition for a constant in pallet config.
@@ -478,6 +480,7 @@ impl ConfigDef {
                        where_clause,
                        attr_span,
                        default_sub_trait,
+                       super_traits: item.supertraits.iter().cloned().collect(),
                })
        }
 }
diff --git a/frame/system/src/lib.rs b/frame/system/src/lib.rs
index 3f6d2274a0b..ae7745df1f7 100644
--- a/frame/system/src/lib.rs
+++ b/frame/system/src/lib.rs
@@ -209,6 +209,7 @@ pub mod pallet {
                        use macro_magic::export_tokens;
                        use sp_runtime::traits::IdentityLookup;
 
+                       #[derive(Eq, Clone, PartialEq)]
                        pub struct TestDefaultConfig {}
 
                        #[export_tokens(TestDefaultConfig)]
```

Why can we not do this? (Really hacky solution for now)